### PR TITLE
Adding names and logging

### PR DESCRIPTION
**Pull request by jean-philippe-martin**
Profiling is easier when the stages have names and output logging information. Keeping my branch as close as possible to 'master' means merges are easier, both ways.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/composite/AddContextDataToRead.java [hunk 1]
```diff
@@ -47,7 +48,7 @@
             assertSameReads(pReads, kvReadRefBases, kvReadVariants);
         }
         // We could add a check that all of the reads in kvReadRefBases, pVariants, and pReads are the same.
-        PCollection<KV<UUID, Iterable<Variant>>> UUIDVariants = kvReadVariants.apply(ParDo.of(new DoFn<KV<GATKRead, Iterable<Variant>>, KV<UUID, Iterable<Variant>>>() {
+        PCollection<KV<UUID, Iterable<Variant>>> UUIDVariants = kvReadVariants.apply(ParDo.named("UUIDVariants").of(new DoFnWLog<KV<GATKRead, Iterable<Variant>>, KV<UUID, Iterable<Variant>>>("UUIDVariants") {
```
Comment (droazen) [inline src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/composite/AddContextDataToRead.java]: Should these names for the `ParDos` (here and below) match the names in the `setName()` calls on the resulting `PCollections`

**Comment by droazen** [comment]
:+1: One minor comment/question, then feel free to merge

File: src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/composite/AddContextDataToRead.java (hunk 1 above)
Comment (jean-philippe-martin) [inline src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/composite/AddContextDataToRead.java]: Good idea. I've updated the names.

**Comment by jean-philippe-martin** [comment]
squashed&rebased. Merge pending tests.

**Pull request merged**